admin/src/main/java/org/bahmni/module/admin/observation/handler/Form2CSVObsHandler.java

```diff
@@ -57,6 +61,39 @@ public Form2CSVObsHandler(CSVObservationHelper csvObservationHelper, FormFieldPa
         return form2Observations;
     }
 
+    @Override
+    public List<EncounterTransaction.Observation> handle(EncounterRow encounterRow, boolean shouldPerformForm2Validations) throws ParseException {
+        if(!shouldPerformForm2Validations)
+            return handle(encounterRow);
+        List<EncounterTransaction.Observation> form2Observations = new ArrayList<>();
+        List<KeyValue> form2CSVObservations = getRelatedCSVObs(encounterRow);
+        for (KeyValue form2CSVObservation : form2CSVObservations) {
+            final List<String> form2CSVHeaderParts = getCSVHeaderPartsByIgnoringForm2KeyWord(form2CSVObservation);
+            if (isNotBlank(form2CSVObservation.getValue())) {
+                verifyCSVHeaderHasConcepts(form2CSVObservation, form2CSVHeaderParts);
+                csvObservationHelper.verifyNumericConceptValue(form2CSVObservation, form2CSVHeaderParts);
+                boolean isMultiSelectObs = formFieldPathService.isMultiSelectObs(form2CSVHeaderParts);
+                boolean isAddmoreConceptObs = formFieldPathService.isAddmore(form2CSVHeaderParts);
+                if(isMultiSelectObs) {
+                    processMultiSelectObs(encounterRow, form2Observations, form2CSVObservation, form2CSVHeaderParts);
+                } else if(isAddmoreConceptObs) {
+                    processAddmoreConcept(encounterRow, form2Observations, form2CSVObservation, form2CSVHeaderParts);
+                } else {
+                    csvObservationHelper.createObservations(form2Observations, encounterRow.getEncounterDate(),
+                            form2CSVObservation, getConceptNames(form2CSVHeaderParts));
+                    setFormNamespaceAndFieldPath(form2Observations, form2CSVHeaderParts);
+                    validateObsForFutureDate(form2Observations, form2CSVObservation, form2CSVHeaderParts);
+                }
+            } else {
+                boolean mandatoryFieldMissing = formFieldPathService.isMandatory(form2CSVHeaderParts);
+                if(mandatoryFieldMissing) {
+                    throw new APIException(format("Empty value provided for mandatory field %s", form2CSVHeaderParts.get(form2CSVHeaderParts.size()-1)));
+                }
+            }
+        }
```

Comment:
I feel this code could be made easier to read. 

Like verifyForMultiSelect(), verifyForAddMore(), verifyForMandatory(). And all the check and verification is embedded inside those methods. Right now this method has too many if-else-if-else in one go. Please try simplifying this. I feel some of the elses are also not needed. Some lines will return false automatically and don't need perf optimsation with if-else.

Reply:
@gsluthra Thanks for review. I have refactored the method.